src/geoaxis.jl

```diff
@@ -55,74 +58,282 @@ display(fig)
     return ax
 end
 
+function draw_geoticks!(ax::Axis, hijacked_observables, line_density)
+    topscene = ax.blockscene
+    scene = ax.scene
 
-function Makie.apply_transform(f::Proj4.Transformation, pt::Point{N,T}) where {N,T}
-    Point(f(Vec(pt)) ./ 100_000)
-end
-function Makie.apply_transform(f::Proj4.Transformation, r::Rect2{T}) where {T}
-    # TODO: once Proj4.jl is updated to PROJ 8.2, we can use
-    # proj_trans_bounds (https://proj.org/development/reference/functions.html#c.proj_trans_bounds)
-    N = 21
-    umin = vmin = T(Inf)
-    umax = vmax = T(-Inf)
-    xmin, ymin = minimum(r)
-    xmax, ymax = maximum(r)
-    for x in range(xmin, xmax; length = N)
-        for y in range(ymin, ymax; length = N)
-            u, v = Makie.apply_transform(f, Point(x, y))
-            umin = min(umin, u)
-            umax = max(umax, u)
-            vmin = min(vmin, v)
-            vmax = max(vmax, v)
+    decorations = Dict{Symbol, Any}()
+
+    xgridpoints = Observable(Point2f[])
+    ygridpoints = Observable(Point2f[])
+
+    xtickpoints = Observable(Point2f[])
+    ytickpoints = Observable(Point2f[])
+
+    xticklabels = Observable(String[])
+    yticklabels = Observable(String[])
+
+    topspinepoints = Observable(Point2f[])
+    btmspinepoints = Observable(Point2f[])
+    lftspinepoints = Observable(Point2f[])
+    rgtspinepoints = Observable(Point2f[])
+
+    xlimits = Observable((0.0f0, 0.0f0))
+    ylimits = Observable((0.0f0, 0.0f0))
+    # First we establish the spine points
+
+    lift(ax.finallimits, ax.xticks, ax.xtickformat, ax.yticks, ax.ytickformat, ax.scene.px_area) do limits, xticks, xtickformat, yticks, ytickformat, pxarea
+
+        lmin = minimum(limits)
+        lmax = maximum(limits)
+        xlimits[] = (lmin[1], lmax[1])
+        ylimits[] = (lmin[2], lmax[2])
+
+        _xtickvalues, _xticklabels = Makie.MakieLayout.get_ticks(xticks, identity, xtickformat, xlimits[]...)
+        _ytickvalues, _yticklabels = Makie.MakieLayout.get_ticks(yticks, identity, ytickformat, ylimits[]...)
+
+        # update but do not notify
+        # project xtickpoints from inputspace to scene dataspace (do transform)
+        # then project to scene pixelspace.
+        # TODO this is still not working right - the text positions almost look stretched
+        # or multiplied out from where they should be.
+        xtickpoints.val = Point2f.(
+            Makie.project.(
+                # obtain the camera of the Scene which will project to its screenspace
+                Ref(camera(scene)),
+                # go from dataspace (transformation applied to inputspace) to pixelspace
+                Ref(:data), Ref(:pixel),
+                # apply the transform to go from inputspace to dataspace
+                Makie.apply_transform(
+                    scene.transformation.transform_func[],
+                    Point2f.(_xtickvalues, ylimits[][1])
+                )
+            )
+        ) .+ Ref(Point2f(pxarea.origin))
```

Comment:
@SimonDanisch I would appreciate some help on this if you have the time - basically, I am trying to project the tick coordinates (which I get in input (lat/long) space from `get_ticks`), to dataspace and then back to `ax.scene`'s pixel space.  Then, I move them to `ax.blockscene`'s pixel space by adding `pixelarea(ax.scene)[].origin`.

I'm not sure if this entire lift statement is just not triggering when it's supposed to, or if something else is going on, but it looks like the ticks are somehow placed at an old position?

Reply:
Nevermind, found the issue - I didn't update the ticklabel positions, assuming they would be read when the ticklabels updated, but I guess that wasn't the case :(